weblate/trans/views/dictionary.py

```diff
@@ -310,11 +310,28 @@ def show_dictionary(request, project, lang):
 
     letterform = LetterForm(request.GET)
 
+    searchform = OneWordForm(request.GET)
+
+    query_string = ''
+
+    # print(searchform.is_valid())
+
+    if searchform.is_valid() and searchform.cleaned_data['term'] != '':
+        # print(searchform.cleaned_data['term'])
+        words = words.filter(
+            source__icontains=searchform.cleaned_data['term']
+        )
+        search = searchform.cleaned_data['term']
+        query_string = 'term={}'.format(search)
+    else:
+        search = ''
+
     if letterform.is_valid() and letterform.cleaned_data['letter'] != '':
         words = words.filter(
             source__istartswith=letterform.cleaned_data['letter']
         )
         letter = letterform.cleaned_data['letter']
+        query_string = 'letter={}'.format(letter)
```

Comment:
This would break when using search *and* letter together, what is probably not what users would expect.

Reply:
What would you suggest? I tried it on running server and it either takes one or another. Previously i was asking if it's necessary to have both methods.
Thanks in advence!

Reply:
You can build the query parameters gradually in a dict and then build whole query string in the end by django.utils.http.urlencode. That will address urlencoding as well.